src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/builders/ConstantsSummaryBuilder.java

```diff
@@ -132,56 +127,55 @@ protected void buildConstantSummary() throws DocletException {
     }
 
     /**
-     * Build the list of packages.
+     * Builds the list of contents for the groups of packages appearing in the constants summary page.
      */
     protected void buildContents() {
         Content contentList = writer.getContentsHeader();
-        printedPackageHeaders.clear();
+        packageGroupHeadings.clear();
         for (PackageElement pkg : configuration.packages) {
-            if (hasConstantField(pkg) && !hasPrintedPackageIndex(pkg)) {
-                writer.addLinkToPackageContent(pkg, printedPackageHeaders, contentList);
+            String abbrevPackageName = getAbbrevPackageName(pkg);
+            if (hasConstantField(pkg) && !packageGroupHeadings.contains(abbrevPackageName)) {
+                writer.addLinkToPackageContent(abbrevPackageName, contentList);
+                packageGroupHeadings.add(abbrevPackageName);
             }
         }
         writer.addContentsList(contentList);
     }
 
     /**
-     * Build the summary for each documented package.
+     * Builds the summary for each documented package.
      *
      * @throws DocletException if there is a problem while building the documentation
      */
     protected void buildConstantSummaries() throws DocletException {
-        printedPackageHeaders.clear();
+        packageGroupHeadings.clear();
         Content summaries = writer.getConstantSummaries();
         for (PackageElement aPackage : configuration.packages) {
             if (hasConstantField(aPackage)) {
                 currentPackage = aPackage;
                 //Build the documentation for the current package.
-
                 buildPackageHeader(summaries);
                 buildClassConstantSummary();
-
-                first = false;
             }
         }
         writer.addConstantSummaries(summaries);
     }
 
     /**
-     * Build the header for the given package.
+     * Builds the header for the given package.
      *
      * @param target the content to which the package header will be added
      */
     protected void buildPackageHeader(Content target) {
-        PackageElement abbrevPkg = configuration.workArounds.getAbbreviatedPackageElement(currentPackage);
-        if (!printedPackageHeaders.contains(abbrevPkg)) {
-            writer.addPackageName(currentPackage, target, first);
-            printedPackageHeaders.add(abbrevPkg);
+        String abbrevPkgName = getAbbrevPackageName(currentPackage);
+        if (!packageGroupHeadings.contains(abbrevPkgName)) {
+            writer.addPackageGroup(abbrevPkgName, target);
```

Comment:
This line is the fix for the "typo" that I referred to in item 1 of the main description.